static_website/main.tf

```diff
@@ -0,0 +1,144 @@
+module "info" {
+  source        = "../info"
+  region        = "${var.region}"
+  environment   = "${var.environment}"
+  account       = "${var.account}"
+}
+
+provider "aws" {
+  region = "${var.region}"
+}
+
+resource "aws_route53_record" "root-domain" {
+  zone_id = "${module.info.hosted_zone_id}"
+  name    = "${var.origin_bucket}.${var.environment}"
+  type    = "A"
+
+  alias {
+    name    = "${aws_cloudfront_distribution.cdn.domain_name}"
+    zone_id = "${aws_cloudfront_distribution.cdn.hosted_zone_id}"
+    evaluate_target_health = false
+  }
+
+}
+
+resource "aws_s3_bucket" "origin" {
+  bucket  = "${var.origin_bucket}-${var.environment}"
+  acl     = "public-read"
+  policy  = <<EOF
+{
+  "Version": "2008-10-17",
+  "Statement": [
+    {
+      "Sid": "PublicReadForGetBucketObjects",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::cloudfront:user/CloudFront Origin Access Identity ${aws_cloudfront_origin_access_identity.origin_access_identity.id}"
+      },
+      "Action": "s3:GetObject",
+      "Resource": "arn:aws:s3:::${var.origin_bucket}-${var.environment}/*"
+    }
+  ]
+}
+EOF
+
+  website {
+    index_document = "${coalesce(var.index_document, "index.html")}"
+    error_document = "${coalesce(var.error_document, "404.html")}"
+    routing_rules  = <<EOF
+[
```

Comment:
This condition feels redundant, shouldn't the *index_document* directive achieve precisely that ?